OpenMeteo/fetch_OpenMeteo_data.py

```diff
@@ -0,0 +1,120 @@
+import openmeteo_requests  # Importing required libraries
+import pandas as pd
+import requests_cache
+from retry_requests import retry
+
+
+class WeatherDataFetcher:
+    BASE_URL = "https://api.open-meteo.com/v1/"  # Base URL for OpenMeteo API
+
+    def __init__(self):
+        # Initialize the WeatherDataFetcher class
+        # Setup the Open-Meteo API client with cache and retry on error
+        cache_session = requests_cache.CachedSession(".cache", expire_after=3600)
+        retry_session = retry(cache_session, retries=5, backoff_factor=0.2)
+        self.openmeteo = openmeteo_requests.Client(session=retry_session)
+
+    def fetch_forecast_data(self, NWP, params):
```

Comment:
```suggestion
    def fetch_forecast_data(self, nwp, params):
```
We don't want to hard code the NWP that we are using. Ideally, we also want type hints for the inputs and outputs.

Reply:
Understood, I'll add type hints as per your suggestion.